addons/hr_expense/models/hr_expense.py

```diff
@@ -783,6 +787,7 @@ def _prepare_payments_vals(self):
             'amount': self.total_amount_currency,
             'payment_type': 'outbound',
             'partner_type': 'supplier',
+            'partner_id': self.vendor_id.id,
```

Comment:
Why do we need to set `partner_id` on both the move and the move lines?

Reply:
To follow the correct payment flow (as here we override everything)
So for reconciliation in this case